File: Python/Queue.py
class ListQueue:
    def __init__(self) -> None:
        self.values = []
    
    def enqueue(self, value):
        self.values.insert(0, value)
    
    def dequeue(self):
        return self.values.pop()
    
    def peek(self):
        if len(self.values) > 0:
            return self.values[-1]
        return None
    
    def size(self):
        return len(self.values)

File: JavaScript/Queues/Queue.mjs
import {LinkedListTail} from '../LinkedLists/LinkedListTail.mjs';

export class Queue{
    constructor(){
        this.values = new LinkedListTail();
        this.length = 0;
    }

    enqueue(data){
        this.values.add(data);
        this.length++;
    }
    dequeue(){
        if (this.length > 0){
            this.length--;
            return this.values.remove();
        }
    }
    peek(){
        return this.values.head.data;
    }
    printList(){
        this.values.printList();
    }

}
